### PR TITLE
fix: compact headings

### PR DESCRIPTION
**Pull request by kellyjosephprice**
| [![PR App][icn]][demo] |
| :--------------------: |

## 🧰 Changes

Fixes a very specific edge case with our compact headings:

```
##Heading
---
```

Should get render as:
```
<h2>Heading</h2>
<hr>
```

But would get render as:

```
<h2>##Heading</h2>
```

Also, to try and prevent this, does a little error checking when configuring the tokenizers.

## 🧬 QA & Testing

- [Broken on production][prod].
- [Working in this PR app][demo].

[demo]: https://markdown-pr-PR_NUMBER.herokuapp.com
[prod]: https://SUBDOMAIN.readme.io
[icn]: https://user-images.githubusercontent.com/886627/160426047-1bee9488-305a-4145-bb2b-09d8b757d38a.svg

### CODE REVIEW

File: __tests__/flavored-parsers/compact-headings.test.js [hunk 1]
```diff
@@ -26,4 +26,15 @@ describe('Compact headings', () => {
     expect(tree.children[0].position.start.offset).toBe(0);
     expect(tree.children[0].position.end.offset).toBe(21);
   });
+
+  it('can parse a compact heading followed by a setext heading', () => {
```
Comment (rafegoldberg) [inline __tests__/flavored-parsers/compact-headings.test.js:30]: Good call to add a test case for this.

File: processor/parse/compact-headings.js [hunk 2]
```diff
@@ -15,12 +17,11 @@ function tokenizer(eat, value) {
...
 function parser() {
-  const { Parser } = this;
-  const tokenizers = Parser.prototype.blockTokenizers;
-  const methods = Parser.prototype.blockMethods;
-
-  tokenizers.compactHeading = tokenizer;
-  methods.splice(methods.indexOf('newline'), 0, 'compactHeading');
+  insertBlockTokenizerBefore.call(this, {
+    name: 'compactHeading',
+    before: 'atxHeading',
```
Comment (rafegoldberg) [inline processor/parse/compact-headings.js:22]: Ahhhh, interesting…

File: processor/parse/utils.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  methods.splice(index, 0, name);
+}
+
+export function insertBlockTokenizerBefore(args) {
+  insertTokenizerBefore.call(this, args);
+}
+
+export function insertInlineTokenizerBefore(args) {
+  return insertTokenizerBefore.call(this, { ...args, type: 'inline' });
+}
```
Comment (rafegoldberg) [inline processor/parse/utils.js:21]: I think we should add a test for this if we're centralizing all the `.splice` logic in to a single place

**Review by rafegoldberg — APPROVED** [review]
I have [one request](https://github.com/readmeio/markdown/pull/801#discussion_r1279899510), but otherwise I think the logic and polish herein is great!

File: processor/parse/utils.js (hunk 3 above)
Comment (kellyjosephprice) [reply processor/parse/utils.js:21]: Hmm, like test this specific function?

**Comment by rafegoldberg** [comment]
## This PR was released!

🚀 Changes included in [<b>v6.65.3</b>](/readmeio/markdown/releases/tag/v6.65.3)
<br>

**Pull request merged**